### DISCUSSION TITLE
Two mesh domain NS Solution

### DISCUSSION THREAD

**Discussion by TLWise** [idaholab/moose#17925] [Q&A Modules: Navier-Stokes] [answered]
I have created a NS model that is composed of two separate mesh domains. The flow for one domain (domain-1) is pressure driven between the inlet and the outlet, walls are treated as no-slip boundary conditions. The second mesh domain is contained in the interior of the first and has different material properties. The problem is transient. I would like to implicitly define within the domain-1  the a no-slip boundary condition based on the outer boundary shape on domain-2. The expectation is that I will be able to observe flow around domain-2 in domain-1. I would then like to translate the viscous stress in domain-1 at the implicit boundary of domain-2 as a boundary force condition in domain-2. This is all done within a single application. Can this be accomplished as described and if so how?
[FallingBall.e.txt](https://github.com/idaholab/moose/files/6541719/FallingBall.e.txt)
[FallingBall.i.txt](https://github.com/idaholab/moose/files/6541720/FallingBall.i.txt)

**Comment by GiudGiud** [comment]
Hello

Do you want to be tracking the interface explicitly or use some sort of volume averaging?
We laid out an approach in this [discussion](https://github.com/idaholab/moose/discussions/17470)
@fdkong this is another fsi-like problem where the solid block is moving around in the fluid. We should make an example of this.

Best,
Guillaume

**Reply by TLWise** [reply]
I would like to track the effects explicitly in each time step iteration. Ball position establishes moving no-slip boundary for the larger fluid domain. The fluid stress at the ball establishes changing ball position.

Thank you.

**Reply by GiudGiud** [reply]
Ok this is very similar to the thread I linked to.
So same advice, do you have a satisfactory solution without the ball moving?
And you'll need to set up elastic mesh deformation, to adapt the mesh at every time step

**Reply by TLWise** [reply]
> Ok this is very similar to the thread I linked to.
> 
> So same advice, do you have a satisfactory solution without the ball moving?
> 
> And you'll need to set up elastic mesh deformation, to adapt the mesh at every time step

Yes, I get very good convergence of the flow solution in the channel. The solution is just absent the blockage and no slip velocity condition that I want to provide by the sphere.

Thank you again for your advise and help.

**Comment by fdkong** [comment, marked as answer]
@TLWise Is domain 2 moving? Does domain 2 deform? 

If so, it seems like an actual FSI example.

**Reply by TLWise** [reply]
Thank you for your interest. Domain 2 is not intended to deform, but is expected to move due to forces from domain 1 and vise versa domain 1 to respond to position of domain 2. Almost fsi but without ALE (mesh movement).

**Reply by fdkong** [reply]
How far domain 2 needs to move? The movement is relatively small or large? It will determine what method we could use.

ALE is referred to as how do we define the fluid domain. Even the domain 2 is rigid, ALE still can be used in domain 1 because domain 2 is moving. One issue with the ALE is that we need to remesh domains  if domain 2 is moving too much (i.e., from the inlet to the outlet). Remeshing is not easy in general, especially in parallel. 

An alternative way might be immersed boundary method where we do not need to remesh and a fixed mesh is used. But we might lose the interpolation accuracy between domain 1 and domain 2.

What equations are you using to define the movement of domain 2 because of the stress from domain 1?  Is it a simple relation?

**Reply by TLWise** [reply]
The immersed boundary method appears to be the solution that I am looking to implement. Domain 2 is allowed to move to any extent within the boundaries of domain 1, but is to be constrained to not cross out of the boundaries of domain 1.

**Reply by TLWise** [reply]
The only other influence on domain 2 outside fluid stress is gravity force.

**Reply by GiudGiud** [reply]
@fdkong Immersed boundary method is like a VOF method right?
Ok for domain 2 staying in domain 1, but do you want the domain 2 mesh to move, or do you want to just be tracking relative fractions of 2 and 1 in a fixed mesh?

**Reply by TLWise** [reply]
Thank you for your continued advice and support with this. I would like to observe the motion of domain 2 within domain one. I have been able to run the model, but I have not observed the effects of each domain on the flow and position solutions.

**Reply by TLWise** [reply]
In the model that I have made. The particle mesh already exist as a subdomain as well as the fluid subdomain. The fsi formulation that I am basing my model on uses the SubdomainBoundingBoxGenerator which appears to constrain me to portions of a square or rectangular subdomain vise the circle that I am trying to track. Also is it required that I specify a bounding box coordinate in SubdomainBoundingBoxGenerator since the subdomains are already defined?

**Reply by GiudGiud** [reply]
Any progress on this?
SubdomainBoundingBoxGen might not be the one you need here. You should have the two blocks defined in the mesh before loading it in MOOSE.

**Reply by TLWise** [reply]
Thank you for remaining engaged in assist with the issue. I had some success using the bounding box approach. What I found in using it, is that the effects of the motion of the bounded region on the mesh deforms the mesh similar to two separate mesh domains with one having motion internal to the other. I need to determine how to do at least two things:

1. Define a bounding shape (circle, oval, octagon, etc.) other than a rectangle.
2. Redefine the bounding shape in the next time step based on the new translation coordinates of the previous time step.

**Reply by GiudGiud** [reply]
I think you will have to add a new mesh generator to do 1. If you could make a parsed expression one I think that would be more valuable than just adding circle. Let us know if you need help with that. 

2. might be tricky. @fdkong do you know how to have the mesh be reset at every time step and mesh generators be ran again?
